include/rocksdb/env_encryption.h

```diff
@@ -194,13 +196,235 @@ class CTREncryptionProvider : public EncryptionProvider {
                                           size_t blockSize);
 
   // CreateCipherStreamFromPrefix creates a block access cipher stream for a
-  // file given given name and options. The given prefix is already decrypted.
+  // file given
+  // given name and options. The given prefix is already decrypted.
   virtual Status CreateCipherStreamFromPrefix(
       const std::string& fname, const EnvOptions& options,
       uint64_t initialCounter, const Slice& iv, const Slice& prefix,
       std::unique_ptr<BlockAccessCipherStream>* result);
 };
 
+class EncryptedSequentialFile : public SequentialFile {
+ private:
+  std::unique_ptr<SequentialFile> file_;
+  std::unique_ptr<BlockAccessCipherStream> stream_;
+  uint64_t offset_;
+  size_t prefixLength_;
+
+ public:
+  // Default ctor. Given underlying sequential file is supposed to be at
+  // offset == prefixLength.
+  EncryptedSequentialFile(SequentialFile* f, BlockAccessCipherStream* s,
```

Comment:
I am concerned about the raw pointers here.  This makes it unclear who owns the CipherStream and SequentialFile.  What happens if there are errors thrown while creating these?  A unique or shared pointer would be a better option,

Reply:
It looks like this was based upon the existing SequentialFileWrapper class in env.h.  That wrapper class still takes raw pointer instead of unique_ptr.  Yes, it too could be updated.